User_Story#1/SortingApp/BusinessLayer/CollectionSortHandler.cs

```diff
@@ -0,0 +1,156 @@
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        private readonly IDataParser<T> parser;
+
+        /// <summary>
+        /// Holds the reader.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        private readonly DataReader reader;
+
+        /// <summary>
+        /// Holds the sorter.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        private CollectionSorterBase<T> sorter;
+
+        /// <summary>
+        /// Holds the writer.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        private readonly DataWriter<T> writer;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CollectionSortHandler{T}"/> class.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        /// <param name="reader">The reader instance.</param>
+        /// <param name="writer">The writer instance.</param>
+        /// <param name="sorter">The sorter instance.</param>
+        /// <param name="parser">The parser instance.</param>
+        public CollectionSortHandler(DataReader reader, DataWriter<T> writer, IDataParser<T> parser)
+        {
+            if (reader != null && writer != null && parser != null)
+            {
+                this.parser = parser;
+                this.reader = reader;
+                this.writer = writer;
+            }
+            else
+                throw new ArgumentNullException();
+        }
+
+        /// <summary>
+        /// Executes sorting for collection.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        public void Execute()
+        {
+            try
+            {
+                this.SortedCollection = new T[this.UnSortedCollection.Length];
+                this.UnSortedCollection.CopyTo(this.SortedCollection, 0);
+                this.sorter.Sort(this.SortedCollection);
+            }
+            catch (ArgumentNullException e)
+            {
+                throw new Exception(e.Message + " was empty.");
+            }
+        }
+
+        /// <summary>
+        /// Generates the sorter.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        /// <param name="type">The type.</param>
+        public void GenerateSorter(SortType type)
+        {
+            switch (type)
+            {
+                case SortType.InsertionSort:
+                    this.sorter = new InsertionSortCreator<T>().Create();
+                    break;
+                case SortType.SelectionSort:
+                    this.sorter = new SelectionSortCreator<T>().Create();
+                    break;
+                case SortType.QuickSort:
+                    this.sorter = new QuickSortCreator<T>().Create();
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// Executes reading for collection.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        public void Read(string path)
```

Comment:
In each public method we should validate all parameters to avoid unexpecting behavior.
Please add validation for empty string, null, string with whitespaces.